src/utils/conversion.ts

```diff
@@ -4067,3 +4074,189 @@ export function inputStatTypeToMeshStatType(
   }
   return statisticsOpTypeToStatType({ op, claimIssuer }, context);
 }
+
+/**
+ * @hidden
+ */
+export function metadataSpecToMeshMetadataSpec(
+  specs: MetadataSpec,
+  context: Context
+): PolymeshPrimitivesAssetMetadataAssetMetadataSpec {
+  const { url, description, typeDef } = specs;
+
+  const {
+    polymeshApi: {
+      consts: {
+        asset: { assetMetadataTypeDefMaxLength },
+      },
+    },
+  } = context;
+
+  const metadataTypeDefMaxLength = u32ToBigNumber(assetMetadataTypeDefMaxLength);
+
+  if (typeDef && metadataTypeDefMaxLength.lt(typeDef.length)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: '"typeDef" length exceeded for given Asset Metadata spec',
+      data: {
+        maxLength: metadataTypeDefMaxLength,
+      },
+    });
+  }
+
+  return context.createType('PolymeshPrimitivesAssetMetadataAssetMetadataSpec', {
+    url: optionize(stringToBytes)(url, context),
+    description: optionize(stringToBytes)(description, context),
+    typeDef: optionize(stringToBytes)(typeDef, context),
+  });
+}
+
+/**
+ * @hidden
+ */
+export function meshMetadataSpecToMetadataSpec(
+  rawSpecs?: Option<PolymeshPrimitivesAssetMetadataAssetMetadataSpec>
+): MetadataSpec {
+  const specs: MetadataSpec = {};
+
+  if (rawSpecs && rawSpecs.isSome) {
+    const { url: rawUrl, description: rawDescription, typeDef: rawTypeDef } = rawSpecs.unwrap();
+
+    if (rawUrl.isSome) {
+      specs.url = bytesToString(rawUrl.unwrap());
+    }
+
+    if (rawDescription.isSome) {
+      specs.description = bytesToString(rawDescription.unwrap());
+    }
+
+    if (rawTypeDef.isSome) {
+      specs.typeDef = bytesToString(rawTypeDef.unwrap());
+    }
+  }
+  return specs;
+}
+
+/**
+ * @hidden
+ */
+export function metadataToMeshMetadataKey(
+  type: MetadataType,
+  id: BigNumber,
+  context: Context
+): PolymeshPrimitivesAssetMetadataAssetMetadataKey {
+  const rawId = bigNumberToU64(id, context);
+
+  let metadataKey;
+  if (type === MetadataType.Local) {
+    metadataKey = { Local: rawId };
+  } else {
+    metadataKey = { Global: rawId };
+  }
+
+  return context.createType('PolymeshPrimitivesAssetMetadataAssetMetadataKey', metadataKey);
+}
+
+/**
+ * @hidden
+ */
+export function meshMetadataValueToMetadataValue(
+  rawValue: Option<Bytes>,
```

Comment:
should these be passed as a single object?

Reply:
Since these were separate attributes every where else, I figured there wasn't that much of a need to put them as an object. Eventually we need destruct them here itself and this is used from one place only